vortex-expr/src/scope_vars.rs

```diff
@@ -0,0 +1,56 @@
+use std::any::{Any, TypeId};
+use std::hash::{BuildHasherDefault, Hasher};
+
+use vortex_utils::aliases::hash_map::HashMap;
+
+/// A TypeMap based on `https://docs.rs/http/1.2.0/src/http/extensions.rs.html#41-266`.
+pub(crate) type ScopeVars = HashMap<TypeId, Box<dyn ScopeVar>, BuildHasherDefault<IdHasher>>;
+
+/// With TypeIds as keys, there's no need to hash them. They are already hashes
+/// themselves, coming from the compiler. The IdHasher just holds the u64 of
+/// the TypeId, and then returns it, instead of doing any bit fiddling.
+#[derive(Default)]
+pub(super) struct IdHasher(u64);
+
+impl Hasher for IdHasher {
+    #[inline]
+    fn finish(&self) -> u64 {
+        self.0
+    }
+
+    fn write(&mut self, _: &[u8]) {
+        unreachable!("TypeId calls write_u64");
+    }
+
+    #[inline]
+    fn write_u64(&mut self, id: u64) {
+        self.0 = id;
+    }
+}
+
```

Comment:
for future reference, there is a popular crate for this already: https://crates.io/crates/nohash-hasher

this is what rerun uses for their typemap https://github.com/emilk/egui/blob/main/crates/egui/src/util/id_type_map.rs